### PR TITLE
Object store/ aws

### PR DESCRIPTION
**Pull request by deepsidhu85**
## Description of changes
What did you change in this pull request?  Provide a description of files changed, user interactions changed, etc.  Include how to test your changes.

Added Amazon AWS S3 object store support.

## Related issue
Link to the GitHub issue this pull request addresses using the `#issuenum` format.  If it completes an issue, use `Fixes #issuenum` to automatically close the issue.

Related to #228 

## Checklist
Things for the developer to confirm they've done before the PR should be accepted:

~* [ ] CHANGELOG.md (and UPGRADING.md if necessary) updated with information for new change.~
~* [ ] Tests added (or description of how to test) for any new features.~
~* [ ] User documentation updated for UI or technical changes.~

### CODE REVIEW

**Comment by JeffreyThiessen** [comment]
Code looks very clean :+1:

**Comment by deepsidhu85** [comment]
> Code looks very clean

Thanks! @JeffreyThiessen

File: src/main/java/ca/corefacility/bioinformatics/irida/repositories/filesystem/IridaFileStorageAwsUtilityImpl.java [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+				throw new StorageException(e.getMessage());
+			} catch (Exception e) {
+				logger.error("Unable to get object from aws S3: " + e);
+				throw new StorageException(e.getMessage());
+			}
+
+			return new IridaTemporaryFile(tempFile, tempDirectory);
+		} catch (FileNotFoundException e) {
+			logger.error(e.getMessage());
+			throw new StorageException(e.getMessage());
```
Comment (tom114) [inline src/main/java/ca/corefacility/bioinformatics/irida/repositories/filesystem/IridaFileStorageAwsUtilityImpl.java:75]: In all these `throw new StorageException` lines can you add the throwable cause to the exception?  That way the  `caused by` lines in the stack trace will actually show us where the exception came from.

So like `throw new StorageException("some message", e)`

**Review by tom114 — CHANGES_REQUESTED** [review]
Tested and this seems to be working very well!  A couple changes below please.

File: src/main/java/ca/corefacility/bioinformatics/irida/repositories/filesystem/IridaFileStorageAwsUtilityImpl.java [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+	 */
+	@Override
+	public IridaTemporaryFile getTemporaryFile(Path file) {
+		try {
+			logger.trace("Getting file from aws s3 [" + file.toString() + "]");
+			Path tempDirectory = Files.createTempDirectory("aws-tmp-");
+			Path tempFile = tempDirectory.resolve(file.getFileName().toString());
+
+			try(S3Object s3Object = s3.getObject(bucketName, getAwsFileAbsolutePath(file))) {
+				S3ObjectInputStream s3ObjectInputStream = s3Object.getObjectContent();
```
[the remark this answers is not in the document]
Comment (deepsidhu85) [reply src/main/java/ca/corefacility/bioinformatics/irida/repositories/filesystem/IridaFileStorageAwsUtilityImpl.java]: Updated in 2ff8d49c203b54e1df18e5388736b81378b1323f. Didn't realize I could add multiple things to the try-with-resources block. Thanks

File: src/main/java/ca/corefacility/bioinformatics/irida/repositories/filesystem/IridaFileStorageAwsUtilityImpl.java (hunk 1 above)
Comment (deepsidhu85) [reply src/main/java/ca/corefacility/bioinformatics/irida/repositories/filesystem/IridaFileStorageAwsUtilityImpl.java]: Updated in 2ff8d49c203b54e1df18e5388736b81378b1323f

**Review by tom114 — APPROVED** [review]

**Pull request merged**